app/packages/core/src/components/SamplesContainer.tsx

```diff
@@ -134,27 +138,28 @@ function SamplesContainer() {
     []
   );
 
-  const defaultSpacesState = {
-    id: "root",
-    children: [
-      {
-        id: "default-samples-node",
-        children: [],
-        type: "Samples",
-        pinned: true,
-      },
-    ],
-    type: "panel-container",
-    activeChild: "default-samples-node",
-  };
+  useEffect(() => {
+    if (!spaces.equals(sessionSpaces)) {
+      updateSpaces(sessionSpaces);
+    }
+  }, [sessionSpaces]);
+
+  useEffect(() => {
+    if (!isMounted.current) {
+      isMounted.current = true;
+      return;
+    }
+    const serializedSpaces = spaces.toJSON();
+    if (!spaces.equals(sessionSpaces) && !spaces.equals(oldSpaces.current)) {
+      setSessionSpaces(serializedSpaces);
+    }
+    oldSpaces.current = serializedSpaces;
+  }, [spaces]);
```

Comment:
I think the solution here is to move this update into `useStateUpdate` so updates are atomic. If it works, then this is ok but I think it would be good to leave a todo or make an issue.

Reply:
The spaces package does not expose directly expose state atoms/selectors. It only exposes the hook which can be used to update the state. I attempted to use those hooks however it did not like that